plugins/database/src/low_level_odbc.cpp

```diff
@@ -536,6 +537,7 @@ _cllExecSqlNoResult(
                  stat, sql );
         result = logPsgError( LOG_NOTICE, icss->environPtr, myHdbc, myHstmt,
                               icss->databaseType );
+        rowCount = -1;
```

Comment:
So we're already in an error `else` block... and we're setting `rowCount = -1` because a calling function expects to see `-1` when there is an error?

But... where is `rowCount` used after this line other than to set `noResultRowCount`... which is ... not used?

Reply:
and what is `result` in this case?   it gets returned below without checking it?

Reply:
Tried to better-ify the comment. Did it work?

Reply:
That's better.

Add a separate commit which does the following:
- Adds a comment to the static global variable explaining its purpose and how it relates to this function and others
- Adds comments throughout this function explaining what it does and why

This is all about helping the next person who enters this function. You probably understand this function better than anyone now and including that knowledge here will keep the next dev from needing to rediscover what you've learned.

Reply:
Added more comments. Are those sufficient, or are there more unclear things?

Reply:
Looks good.